### ISSUE TITLE
Minecraft has a weird red border on macOS

### ISSUE THREAD

**Issue by Mr-Aerod** [FlashyReese/sodium-extra-fabric#297]
### Make sure you are not opening a duplicate.

- [ ] [I have searched the issue tracker](https://github.com/FlashyReese/sodium-extra-fabric/issues?q=) and did not find an issue describing my bug.

### Minecraft version.

1.19.3

### Sodium Extra version.

0.4.16

### Sodium version.

0.4.8

### What happened?

![Snímek obrazovky 2023-01-03 v 14 35 08](https://user-images.githubusercontent.com/79194869/210369068-d6a8b603-4caa-4f55-a9d3-c2e33e0f55f4.png)
Basically every time I start Minecraft with this mod installed, the screen gets resized weirdly like this leaving an empty red area next to it. This ain't the first version that has that bug, it has also been a thing on 1.19.2.
It doesn't happen without this mod and it can be fixed by resizing the window, though its still a bit annoying.

### Relevant log.

[latest.log](https://github.com/FlashyReese/sodium-extra-fabric/files/10337035/latest.log)

**Comment by FlashyReese** [comment]
I think it's probably the `Reduce Resolution on macOS` under `Extras` causing it, you can disable it to see if the issue still persists obviously you will lose out on performance. ~~Can you provide me information on what hardware your mac has and what version of macOS are you running?~~

**Comment by FlashyReese** [comment]
Can you give this build a test to see if the issue persists when having `Reduce Resolution on macOS` enabled?

[sodium-extra-0.4.16+mc1.19.3-unknown.zip](https://github.com/FlashyReese/sodium-extra-fabric/files/10337978/sodium-extra-0.4.16%2Bmc1.19.3-unknown.zip)

**Comment by Mr-Aerod** [comment]
Where do I find that Reduce Resolution setting?

**Comment by Mr-Aerod** [comment]
By the way, I am running latest macOS Ventura through Hackintosh, OpenCore.

**Comment by FlashyReese** [comment]
Well, I'm not entirely sure how Apple's Retina works on Hackintosh. 
![2023-01-04_10 43 46](https://user-images.githubusercontent.com/30311066/210474863-9c93d85d-e277-4a56-937f-3b5e5beb54e3.png)
It can be found on the `Extras` tab in Video Settings.

**Comment by iAdrianRobin** [comment]
> Can you give this build a test to see if the issue persists when having `Reduce Resolution on macOS` enabled?
> 
> [sodium-extra-0.4.16+mc1.19.3-unknown.zip](https://github.com/FlashyReese/sodium-extra-fabric/files/10337978/sodium-extra-0.4.16%2Bmc1.19.3-unknown.zip)

I had the same issue on a MacBook Air (M1, 2020) 8-Core GPU, 16GB RAM, 1TB SSD running macOS 13.2.1 (22D68)
and tested build I had following results:

Szenario: Sodium Extra 0.4.16+mc1.19.2-build.90 "reduces Resolution on macOS" - On
= described Issue

Szenario: Sodium Extra 0.4.16+mc1.19.2-build.90 reduces "Resolution on macOS" - Off
= Issue gone

Szenario: sodium-extra-0.4.16+mc1.19.3-unknown.zip
= Issue gone
But Video Settings only show:
-General
-Quality
-Performance
-Advanced
All other Video settings such as Animations, Particles, Details, Render and Extras are not visible.
So toggling the "Resolution on macOS" Setting is not possible with this build.

**Comment by FlashyReese** [comment]
> Szenario: sodium-extra-0.4.16+mc1.19.3-unknown.zip
= Issue gone
But Video Settings only show:
-General
-Quality
-Performance
-Advanced
All other Video settings such as Animations, Particles, Details, Render and Extras are not visible.
So toggling the "Resolution on macOS" Setting is not possible with this build.

@feldstein-iAR I don't think the mod was loaded properly. Did you install the JAR from the zip? (The one **without** the `-sources.jar` file name). If you did can you send me the game's log here?

**Comment by iAdrianRobin** [comment]
@FlashyReese Yes it seems that I installed it wrong. Now I tested again with a new clean instance just with Sodium installed and no other mods beside "Fabric API", "Reese's Sodium Options" and "Sodium Extra".

for Sodium Extra I tested
0.4.17+mc1.19.3-build.95
vs  
sodium-extra-0.4.16+mc1.19.3-unknown

Now i see all the options but they share all the same behaviour
"reduces Resolution on macOS" - On = described Issue
"Resolution on macOS" - Off = Issue gone

But there is one additional thing I would like to mention that I noticed now while testing. When it has started with this boarder it seems to be a visual glitch only as cursor movement does not align. In fact, it seems that the cursor moves like normal and the buttons get highlited when the mouse hovers over the place they supposed to be, not where they actually appear.

In addition when the game window size gets changed by manual adjusting it even the slightest, or by maximising it and then changed back to window mode everything seems to snap back in to place where it should be and all seems fine.

So maximising to full screen and then back to window mode seems currently a possible work around for all versions i've tested. This also works in my normal setup with other mods installed. Then only the red boot screen would be off center.

Hope this feedback would help you.

**Comment by FlashyReese** [comment]
It appears that the patch didn't work, and unfortunately, I am unable to debug this issue on a Mac since I don't own one. The only workaround for now is to resize the window.

I'm considering whether to disable this feature by default since it impacts the user experience. However, it does provide a significant performance improvement for Mac users.

If I can access a Mac, I'll work on fixing this issue to improve the user experience.

**Comment by mrjasonn** [comment]
> It appears that the patch didn't work, and unfortunately, I am unable to debug this issue on a Mac since I don't own one. The only workaround for now is to resize the window.
> 
> I'm considering whether to disable this feature by default since it impacts the user experience. However, it does provide a significant performance improvement for Mac users.
> 
> If I can access a Mac, I'll work on fixing this issue to improve the user experience.

This is a very good idea. Mentioned in #311.

**Comment by TheFruxz** [comment]
Maybe tell about this bug in the hover message of this option? This would be great and avoid people removing mod-by-mod to find the mod causing this bug.

**Comment by stantios** [comment]
I just tested to see if the bug is still present and for me, this bug is fixed. I don't see a red border anymore when the "Reduce Resolution on macOS" is enabled.

**Comment by TheFruxz** [comment]
> I just tested to see if the bug is still present and for me, this bug is fixed. I don't see a red border anymore when the "Reduce Resolution on macOS" is enabled.

I just checked it again and on my 1080p monitor (haven't checked other displays/native displays yet) the bug is still present on macOS.
Here is a screenshot with the setting disabled after startup:

![SCR-20240511-qdmp](https://github.com/FlashyReese/sodium-extra-fabric/assets/28064149/1f675e9b-9be9-4093-9f48-0ad2e6bcd4f3)

and with the setting enabled:

![SCR-20240511-qdzg](https://github.com/FlashyReese/sodium-extra-fabric/assets/28064149/8a10a10b-127f-4e1a-bdf0-ed93cc075a0a)

**Comment by stantios** [comment]
Weird, I have a 21.5-inch (4096 x 2304) iMac and this issue doesn't affect me. This could affect certain monitors, but I am not sure.

**Comment by Lumysia** [comment]
same issue here

MacBook Pro 16-inch, 2021, Apple M1 Pro, macOS 15.2
Minecraft 1.21.1, NeoForge 21.1.93
sodium-extra-neoforge-0.6.0+mc1.21.1.jar

**Comment by iblowmymind** [comment]
Yeah, same here. It seems Reduce Resolution on macOS on NeoForge is broken entirely, as resizing the window makes the game not run on half resolution anymore. (3024x1890 resolution displayed on the F3 screen when the same configuration on Fabric makes it 1512x945).
So, even though the "red border" is fixed on a window resize it ultimately doesn't change the fact that reduced resolution doesn't work.

**Comment by FilipRank** [comment]
The issue is caused by the neoforge loading screen. You can fix this by setting `earlyWindowControl` to false in `config/fml.toml` . It's unfortunate that nobody else here knew how to help, it's really a simple fix.

**Comment by Dioshat** [comment]
> 问题是由 neoforge 的加载画面引起的。你可以在 中设置为 false 来解决这个问题。很遗憾这里没人知道怎么帮忙，这其实是个简单的修复方法。`earlyWindowControl``config/fml.toml`

You're an absolute genius!!!!!!!!!!!!!!
